core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.tableversions.core
+
+import java.time.Instant
+
+import com.gu.tableversions.core.TableVersions.CommitResult
+
+/**
+  * This defines the interface for querying and updating table version information tracked by the system.
+  */
+trait TableVersions[F[_]] {
+
+  /**
+    * Start tracking version information for given table.
+    * This must be called before any other operations can be performed on this table.
+    */
+  def init(table: TableName): F[Unit]
```

Comment:
Interesting point. My initial thought was that this only needs to be called once per table, e.g. the first time the table is populated (a bit like you call `git init` when you first initialise a git repo). But as we want jobs to auto-create new tables, we'd either need an `exists` method - or just make the `init` method idempotent so it can be called every time the job runs.

But! I can imagine use cases, e.g. in the command line for this, where you may want to call one of the other operations on the version store, where you definitely wouldn't want to call `init` first. E.g. if you run a command like `table-versions log clean.foobarbaz` you would probably want this to return an error saying `No table 'clean.foobarbaz' found`, rather than going and creating it in the version store.

Reply:
You've just made me wonder if `git init` is idempotent 😄 

Yeah, I think some something like:

```scala
def init(name: String): F[Table.Id]
def exists(name: String): F[Option[Table.Id]]
```

pair could be nice. Then all other ops pass this `Table.Id` around.

Reply:
Not sure if having an `Id` type adds much in that case, as the `TableName` already is the unique ID for the table? Happy to discuss anyway!

Reply:
I took it as the only way of getting a `Table.Id` is through calling one of `init` or `exists` and thus this ensures that one has been called before doing the other dependent operations...?

Reply:
Yes, I think that was the idea, but my concern is that there's no guarantee that the table is still there by the time a client calls one of the other operations. I.e. some other client may have deleted the table. So, you have to deal with those cases anyway, in which case I don't know if the extra level of indirection adds much.